### PR TITLE
tech: Create Alert component

### PR DESCRIPTION
**Pull request by Youakeem**
## Describe your changes
Create a composable Alert component which supports different variants and icons.

## Justify why they are needed
To be used to create the InfoCard component with more flexibility with composition.

## Checklist before requesting a review

- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by Youakeem** [comment]
* **#4320** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4320?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Youakeem and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/ui/src/components/Alert/Alert.css.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  defaultVariants: {
+    variant: 'info',
+  },
+})
+
+export const iconStyles = style({
+  marginTop: tokens.space.xxs,
+})
+
+globalStyle(`${iconStyles} path`, {
```
Comment (gustaveen) [inline packages/ui/src/components/Alert/Alert.css.ts:59]: Does this style need to be set globally?

**Review by gustaveen — COMMENTED** [review]
Looks great! Just wondering about if we need to abstract it to a new Alert component? Will we ever use it other than in the InfoCard? Then we could probably just move this into the InfoCard component directly, it's nice if it's a 1 to 1 mapping with the UI kit in Figma IMO

File: packages/ui/src/components/Alert/Alert.css.ts (hunk 1 above)
Comment (Youakeem) [reply packages/ui/src/components/Alert/Alert.css.ts:59]: Yup, not possible to select children unfortunately. I tried using `selectors` but it broke and the terminal said this is how it should be done instead.

**Comment by Youakeem** [comment]
> Looks great! Just wondering about if we need to abstract it to a new Alert component? Will we ever use it other than in the InfoCard? Then we could probably just move this into the InfoCard component directly, it's nice if it's a 1 to 1 mapping with the UI kit in Figma IMO
> 
> Does this style need to be set globally?

Either would work with me. I think of `Alert` as the primitive that supports different compositions (Icon, text, buttons) and `InfoCard`, `AttentionCard` ..etc as abstractions for common use cases (Specific icon, no buttons).

File: packages/ui/src/components/Alert/Alert.css.ts (hunk 1 above)
Comment (gustaveen) [reply packages/ui/src/components/Alert/Alert.css.ts:59]: Ah right, it's a bit of a mind shift to not target childrens :)

Comment (alebedev) [reply packages/ui/src/components/Alert/Alert.css.ts:59]: Can we use color=currentColor in SVG to inherit color from DOM parent instead?

File: packages/ui/src/components/Alert/Alert.css.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  fill: AlertElement,
+})
+
+export const bodyStyles = style({
```
Comment (alebedev) [inline packages/ui/src/components/Alert/Alert.css.ts:63]: It is time to app sprinkle for `flexGrow: 1`?

File: packages/ui/src/components/Alert/Alert.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    variant: {
+      options: ['info', 'warning', 'error', 'success'],
+      control: { type: 'select' },
+    },
+  },
+  args: {
+    variant: 'info',
+  },
+  parameters: {
+    design: {
```
Comment (alebedev) [inline packages/ui/src/components/Alert/Alert.stories.tsx:37]: Nice!

File: packages/ui/src/components/Alert/Alert.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+function Actions({ children, className, ...props }: ActionsProps) {
+  return (
+    <div className={clsx(actionsContainerStyles, className)} {...props}>
+      {children}
+    </div>
+  )
+}
+
+type ActionProps = ButtonProps<'button'>
```
Comment (alebedev) [inline packages/ui/src/components/Alert/Alert.tsx:66]: `ComponentProps` above is a nice touch.  Can you use something like `ComponentProps<typeof Button<'button'>>` and does it look better or worse than just `ButtonProps`?

**Comment by gustaveen** [comment]
> > Looks great! Just wondering about if we need to abstract it to a new Alert component? Will we ever use it other than in the InfoCard? Then we could probably just move this into the InfoCard component directly, it's nice if it's a 1 to 1 mapping with the UI kit in Figma IMO
> > Does this style need to be set globally?
> 
> Either would be fine by me. However, I think of `Alert` as the primitive that supports different compositions (Icon, text, buttons) and `InfoCard`, `AttentionCard` ..etc as abstractions for common use cases (Specific icon, no buttons).

I like that it's composable! 👌 Let's put it in InfoCard then perhaps and we could move InfoCard to ui kit

File: packages/ui/src/components/Alert/Alert.css.ts (hunk 1 above)
Comment (Youakeem) [reply packages/ui/src/components/Alert/Alert.css.ts:59]: Good point, the icon components take a `color` prop which sets `fill` on the `path` and it defaults to `currentColor`. I'll try setting color on the SVG itself.

**Review by alebedev — APPROVED** [review]

File: packages/ui/src/components/Alert/Alert.css.ts (hunk 1 above)
Comment (Youakeem) [reply packages/ui/src/components/Alert/Alert.css.ts:59]: Nice! works beautifully!

File: packages/ui/src/components/Alert/Alert.css.ts (hunk 2 above)
Comment (Youakeem) [reply packages/ui/src/components/Alert/Alert.css.ts:63]: I am thinking about it! I think we should have base sprinkles which is what we have now, and then extend those to create `patterns` like `flexSprinkles` for example.

I'll play with it a little and open a PR.

File: packages/ui/src/components/Alert/Alert.stories.tsx (hunk 3 above)
Comment (Youakeem) [reply packages/ui/src/components/Alert/Alert.stories.tsx:37]: Yes, I think it's a pretty nice addition as well! I got it from another story tho!

File: packages/ui/src/components/Alert/Alert.tsx (hunk 4 above)
Comment (Youakeem) [reply packages/ui/src/components/Alert/Alert.tsx]: They are the same. I've updated to use `ComponentProps`.

**Comment by Youakeem** [comment]
### Merge activity

* **Jun 3, 2:54 AM EDT**: @Youakeem started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4320).

**Pull request merged**